NineAnimator/Models/Anime Source/animesaturn.com/AnimeSaturn.swift

```diff
@@ -0,0 +1,88 @@
+//
+//  This file is part of the NineAnimator project.
+//
+//  Copyright © 2018-2020 Marcus Zhou. All rights reserved.
+//
+//  NineAnimator is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  NineAnimator is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with NineAnimator.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+import Alamofire
+import Foundation
+
+#if canImport(UIKit)
+import UIKit
+#elseif canImport(AppKit)
+import AppKit
+#endif
+
+class NASourceAnimeSaturn: BaseSource, Source, PromiseSource {
+    var name: String { "animesaturn.com" }
+    
+    var aliases: [String] { [] }
+    
+    #if canImport(UIKit)
+    var siteLogo: UIImage { #imageLiteral(resourceName: "AnimeSaturn Site Icon") }
+    #elseif canImport(AppKit)
+    var siteLogo: NSImage { #imageLiteral(resourceName: "AnimeSaturn Site Icon") }
+    #endif
+    var siteDescription: String {
+        "AnimeSaturn è un server italiano. AnimeSaturn is a free website that provides Italian subtitled anime. The website may be region blocked."
+    }
+    
+    class var AnimeSaturnStream: Anime.ServerIdentifier {
+        "AnimeSaturn"
+    }
+    
+    var preferredAnimeNameVariant: KeyPath<ListingAnimeName, String> {
+        \.english
+    }
+    
+    override var endpoint: String { "https://animesaturn.com" }
+    
+    override init(with parent: NineAnimator) {
+        super.init(with: parent)
+        
+        // Setup Kingfisher request modifier
+        setupGlobalRequestModifier()
+        
+        requestManager.enqueueValidation {
+            _, response, _ in
+            if response.statusCode == 403,
+                response.headers["server"]?.hasPrefix("cloudflare") == true {
+                return .failure(NineAnimatorError.contentUnavailableError(
+                    "AnimeSaturn.it is not available in your region. Please use other sources instead."
+                ))
+            }
+            return .success(())
+        }
```

Comment:
Seems like animesaturn.com is available in the other parts of the world. These lines are used in animeunity to detect geoblocks. You can delete these lines if you want.